src/benchmark.rs

```diff
@@ -161,89 +196,136 @@ impl Benchmark {
         .map(Either::IntervalCounter)
         {
             if !counter.is_enabled() {
-                return;
+                continue;
             }
 
-            BENCH.lock().reset(counter);
+            BENCH.lock().reset(&counter, &scope);
         }
     }
 
     /// Stop and record interval counter.
-    pub fn stop_interval_counters(tag: &str) {
+    pub fn stop_interval_counters(scope: Scope) {
         if !config::BENCHMARK {
             return;
         }
 
         for counter in [
             IntervalCounter::ExecutionTime,
             IntervalCounter::InstructionRet,
-        ]
-        .map(Either::IntervalCounter)
-        {
-            if !counter.is_enabled() {
-                return;
+        ] {
+            let wrapped_counter = Either::IntervalCounter(counter);
+
+            if !wrapped_counter.is_enabled() {
+                continue;
             }
 
-            let bench = counter.name();
-            let value = counter.reset_value() - BENCH.lock().reset(counter);
+            let mut bench = BENCH.lock();
+            let value =
+                wrapped_counter.reset_value() - bench.read_interval_counters(&counter, &scope);
 
-            Self::record_entry(bench, tag, value);
+            bench.update_inteval_counter_stats(&counter, &scope, value);
         }
     }
 
+    fn update_inteval_counter_stats(
+        &mut self,
+        counter: &IntervalCounter,
+        scope: &Scope,
+        value: usize,
+    ) {
+        let index = Self::interval_counter_index(counter, scope);
+        let stats = &mut self.interval_counters[index];
+        stats.count += 1;
+        stats.sum += value;
+        stats.mean = stats.sum / stats.count;
+        stats.min = if value < stats.min { value } else { stats.min };
```

Comment:
You can replace those by [`core::cmp::min`](https://doc.rust-lang.org/nightly/core/cmp/fn.min.html), and manual max by `core::cmp::max`.

On some architecture it is possible to have a better implementation than using an `if` (using branch-less instructions such as conditional moves which do not pollute the CPU branch predictor). So using the built-in Rust function gives us the best implementation we can have.

Reply:
Thank you! :)